packages/elements/src/react/common/form/index.tsx

```diff
@@ -148,12 +148,16 @@ const useInput = ({
   const onChange = React.useCallback(
     (event: React.ChangeEvent<HTMLInputElement>) => {
       onChangeProp?.(event);
-      if (!name) return;
+      if (!name || initialValue) return;
       ref.send({ type: 'FIELD.UPDATE', field: { name, value: event.target.value } });
     },
-    [ref, name, onChangeProp],
+    [ref, name, onChangeProp, initialValue],
   );
 
+  React.useEffect(() => {
+    ref.send({ type: 'FIELD.UPDATE', field: { name, value: initialValue } });
```

Comment:
❓ Should we check if name/initialValue are present? No real need to send the event if the initialValue isn't present, right?

```suggestion
    ref.send({ type: 'FIELD.UPDATE', field: { name, value: initialValue } });
```

Reply:
Good callout https://github.com/clerk/javascript/pull/3054/commits/08c60f2bf355881549c20fd68708a03a1d7c71a0